### DISCUSSION TITLE
Can we activate vs code extension through activation event by reading url from particular file

### DISCUSSION THREAD

**Discussion by inchara-1998** [microsoft/vscode-discussions#713] [Extension Development QnA] [answered]
We a config file inside folder .git for any repo
we need to read the url inside that file and activate the extension

is this achievable using activation events or any other ways

**Comment by r3m0t** [comment, marked as answer]
Use a general activation event like `workspaceContains` and read the file in your `activate` function + whenever the file changes. Instead of calling eg `registerCodeLensProvider` from `activate`, you can call it when the condition of the config file URL is met. That way, you will save resources when the user has a window open where they won't want to use your extension.

**Reply by r3m0t** [reply]
`registerCodeLensProvider` was just an example, this applies to any feature you register. You can also use context keys to prevent your commands from appearing in Command Palette in windows they can't be used.
